packages/svelte/src/internal/client/render.js

```diff
@@ -48,6 +48,10 @@ export function set_should_intro(value) {
  * @returns {void}
  */
 export function set_text(text, value) {
+	// For objects, we apply string coercion before diffing
+	if (typeof value === 'object' && value !== null) {
+		value = value + '';
+	}
```

Comment:
I think it was you who taught me that we should avoid reassigning parameters, since mutating the `arguments` object carries overhead. Should it be something like this instead?

```suggestion
	var str = value == null ? '' : typeof value === 'object' ? value + '' : value;
```

Reply:
Good point!

Reply:
> mutating the arguments object carries overhead

That is an interesting observation. Could you point to any sources? I couldn't find any measurable difference between two approaches https://jsben.ch/Op7ml